compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -343,7 +343,12 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
               val jpath = file.jpath
               val relativePath =
                 if jpath eq null then file.path // repl and other custom tests use abstract files with no path
-                else if jpath.isAbsolute then sourcerootPath.relativize(jpath.normalize).toString
+                else if jpath.isAbsolute then
+                  val cunitPath = jpath.normalize
+                  // On Windows we can only relativize paths if root component matches
+                  // (see implementation of sun.nio.fs.WindowsPath#relativize)
+                  try sourcerootPath.relativize(cunitPath).toString
+                  catch case _: IllegalArgumentException => cunitPath.toString
```

Comment:
I'm wondering why it is not caught by the Windows CI. Can we add a test for this in Windows CI?

Reply:
You'll notice it only if your project directory **and** the directory used fir the test suite are different.

As sketched above, my project directory is `W:\scala3-windows\` (`W:` contains my local GitHub repo for [`michelou/dotty-examples`](https://github.com/michelou/dotty-examples/)) while temporary files created by the test suite are located in `C:\Users\michelou\AppData\Local\Temp\` (aka. `%LOCAL_APPDATA%`).

Reply:
Answering your question "Can we add a test for this in Windows CI?"

As a reminder the two variables `LOCALAPPDATA` and `TEMP` have the following *default values* on Windows:
```batch
LOCALAPPDATA=C:\Users\%USERNAME%\AppData\Local
TEMP=%LOCALAPPDATA%\Temp
```
**NB.** With a Github-hosted runner I have `USERNAME=runneradmin`, locally I have `USERNAME=michelou`.

Let's assume we *can* execute command [`subst`](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/subst) in a CI workflow; then it should possible to write something like the following (eg. in batch file `cmdTests.bat`):
```batch
subst X: %LOCALAPPDATA%
set TEMP=X:\Temp
call "%_SBT_CMD%" ";scalac ..."
set TEMP=%LOCALAPPDATA%\Temp
subst /D X:
```
The idea is to reassign the `TEMP` variable to some path with a drive letter different from `C:` (in this case `X:`) *before* running `scalac` with a Tasty related test example (derived from one of the above failing tests).

Reply:
Thanks for the detailed explanation 👍 

I just checked that our Windows CI machine supports the command `subst`. Let me know if you'd like to add a test in this PR or later.